src/main/java/Deadline.java

```diff
@@ -0,0 +1,16 @@
+public class Deadline extends Task {
+    String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    @Override
+    public String getLabel() {
+        String typeIndicator = "[D]";
+        String doneIndicator = "[" + (this.isDone ? "X" : " ") + "]";
+        String suffix = "(by: " + this.by + ")";
```

Comment:
Perhaps you should rename this to something like "dueDateIndicator" to match with the other variable names and make it clearer what this variable is for